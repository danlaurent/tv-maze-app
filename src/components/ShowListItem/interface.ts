import { ISearchedShow, IShow } from '../../interface/shows';

export interface IShowListItem {
  show: IShow;
  onPress?: (showId: number) => void;
}
