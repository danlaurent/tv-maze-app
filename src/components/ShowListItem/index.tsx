import React from 'react';
import { StyleSheet, View, Image, Text, TouchableOpacity } from 'react-native';
import { IShowListItem } from './interface';
import Poster from '../Poster';
import DetailsMainInfo from '../DetailsMainInfo';
import { humanizeMetaInfo } from '../../utils/transform';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
  poster: {
    marginRight: 12,
  },
});

const ShowListItem = ({ show, onPress = () => {}, testID }: IShowListItem) => {
  return (
    <TouchableOpacity
      testID={testID}
      style={styles.container}
      onPress={() => onPress(show.id)}
    >
      <Poster source={show.image?.medium} size='small' style={styles.poster} />
      <DetailsMainInfo
        testID={`${testID}_detailsMainInfo`}
        title={show.name}
        subtitle={humanizeMetaInfo(show.genres)}
        rating={show.rating.average}
        align='flex-start'
      />
    </TouchableOpacity>
  );
};

export default ShowListItem;
