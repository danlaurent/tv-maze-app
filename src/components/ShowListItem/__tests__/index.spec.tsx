import React from 'react';
import { render, fireEvent, RenderAPI } from '@testing-library/react-native';
import ShowListItem from '..';

const showMock: any = {
  name: 'Test Movie',
  rating: {
    average: 8.0,
  },
  image: {
    medium: 'https://via.placeholder.com/150',
  },
  genres: ['Test'],
};

const onPressMock = jest.fn();

describe('ShowListitem', () => {
  let tree: RenderAPI;
  beforeEach(() => {
    tree = render(
      <ShowListItem
        testID='testShowListItem'
        show={showMock}
        onPress={() => onPressMock()}
      />
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const snapshot = tree.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('When the ShowListItem is pressed', () => {
    beforeEach(async () => {
      await fireEvent.press(tree.getByTestId('testShowListItem'));
    });

    it('fires the onPress action', () => {
      expect(onPressMock).toBeCalledTimes(1);
    });
  });
});
