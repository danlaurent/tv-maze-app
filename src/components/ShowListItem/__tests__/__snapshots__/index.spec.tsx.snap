// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowListitem renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
    }
  }
  testID="testShowListItem"
>
  <View
    style={
      Object {
        "borderRadius": 4,
        "height": 147,
        "marginRight": 12,
        "width": 105,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://via.placeholder.com/150",
        }
      }
      style={
        Object {
          "borderRadius": 4,
          "height": 147,
          "width": 105,
        }
      }
      testID="undefined_image"
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "700",
          "marginBottom": 8,
          "marginTop": 16,
        }
      }
    >
      Test Movie
    </Text>
    <Text
      style={
        Object {
          "color": "#6d6f7e",
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
      testID="testShowListItem_detailsMainInfo_subtitle"
    >
      Test
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "#c6c7d2",
            "fontSize": 16,
            "fontWeight": "700",
            "marginLeft": 8,
          }
        }
        testID="testShowListItem_detailsMainInfo_rating"
      >
        8
      </Text>
    </View>
  </View>
</View>
`;
